### PR TITLE
Better setup errors

### PR DESCRIPTION
**Pull request by Inrixia**
This resolves https://github.com/alvgaona/cloudflare-discord-bot/issues/11

The new response for `/setup` is a json object containing each command and the resulting response from the discord api.
Allowing for easy debugging and full info on the response for each commands setup as can be seen below:
image

If there is a different way you wanted to approach this or changes that you need please just let me know.

### CODE REVIEW

**Review by alvgaona — CHANGES_REQUESTED** [review]
Looks great. Would you mind amending those formatting errors before we merge this? I'll add a tool to the repo soon to avoid this type of problem.

**Comment by Inrixia** [comment]
> Looks great. Would you mind reverting those formatting changes before we merge this? I'll add a tool to the repo soon to avoid this type of problem.

I've setup prettier with formatting settings following what already exists (spaces vs tabs etc) and re pushed changes. Only a few things unrelated to this PR got affected, Ill do the same for https://github.com/alvgaona/cloudflare-discord-bot/pull/13 now.

Let me know if you want any other changes

File: package.json [hunk 1]
```diff
@@ -1,44 +1,41 @@
...
-    "itty-router": "^2.4.2",
-    "tweetnacl": "^1.0.3"
-  },
-  "scripts": {
-    "build": "tsc -p .",
-    "format": "prettier --write  '*.{json,js}' 'src/**/*.{js,ts}' 'test/**/*.{js,ts}'",
-    "lint": "eslint --max-warnings=0 src && prettier --check '*.{json,js}' 'src/**/*.{js,ts}' 'test/**/*.{js,ts}'",
-    "test": "echo \"No tests specified\""
-  }
+	"name": "cloudflare-discord-bot",
```
Comment (alvgaona) [inline package.json:2]: So we're using 4 spaces rather than 2? I don't mind, just want to confirm.

File: src/setup.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import { Buffer } from 'buffer';
+import { Buffer } from "buffer";
```
Comment (alvgaona) [inline src/setup.ts:1]: Double quotes then? Again, it's the same for me.

File: src/setup.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 import { ApplicationCommand, InteractionHandler } from "./types";
 
 const btoa = (value: string) => Buffer.from(value, "binary").toString("base64");
 
 const getAuthorizationCode = async (headers: any) => {
-  headers['Content-Type'] = 'application/x-www-form-urlencoded'
+  headers["Content-Type"] = "application/x-www-form-urlencoded";
```
Comment (alvgaona) [inline src/setup.ts:7]: Semi-colon at the end then? 😛

File: src/setup.ts [hunk 4]
```diff
@@ -50,19 +49,12 @@ const deleteExistingCommands = async (applicationId: string, bearer: any, guildI
...
-        return fetch(
-          `${url}/${command.id}`,
-          {
-            method: "DELETE",
-            headers: { Authorizaton: `Bearer ${bearer}`}
-          }
-        );
-      }
-    )
+    commands.map((command: ApplicationCommand & { id: string; application_id: string }) => {
```
Comment (alvgaona) [inline src/setup.ts:52]: Thank you, looks much better.

**Review by alvgaona — APPROVED** [review]
Thx!

**Comment by Inrixia** [comment]
Gtg will be back later tonight. Happy to modify the prettier config with settings to enforce styling to what youd prefer just let me know what to use :)

**Comment by alvgaona** [comment]
@Inrixia I forgot to mention that you have a few changes that are the same in this other PR #13, e.g., the `.prettierrc.json`.

**Comment by alvgaona** [comment]
> Gtg will be back later tonight. Happy to modify the prettier config with settings to enforce styling to what youd prefer just let me know what to use :)

Cool. Let's decide together. I'll suggest a few rules that are of my liking.

**Comment by Inrixia** [comment]
> @Inrixia I forgot to mention that you have a few changes that are the same in this other PR #13, e.g., the `.prettierrc.json`.

Yep the env branch is actually branched off this one so changes should be reflected as one commit after merging. After finalizing prettier changes here I can modify the branched commit and force push to reflect these changes there.

File: package.json (hunk 1 above)
Comment (Inrixia) [reply package.json:2]: I usually use tabs, i have set the prettier config to use `tabWidth: 2` but maybe it's not respecting it.

If you want to remove any ambiguity I would switch to tabs as thats the default for prettier

File: src/setup.ts (hunk 2 above)
Comment (Inrixia) [reply src/setup.ts:1]: This is a prettier thing, can either default to `"` or `'` but not mixed. I personally would leave it as the default

File: src/setup.ts (hunk 3 above)
Comment (Inrixia) [reply src/setup.ts:7]: Can disable it but since its done automatically I don't think it really matters

File: src/setup.ts (hunk 4 above)
Comment (Inrixia) [reply src/setup.ts:52]: Id take credit but all the formatting cleanup was all handled automatically by prettier :P

**Comment by Inrixia** [comment]
Let me know your thoughts on the prettier changes. After this branch is merged I can open a new PR with the prettier config we decide on applied across all files for consistency. Then https://github.com/alvgaona/cloudflare-discord-bot/pull/13 and https://github.com/alvgaona/cloudflare-discord-bot/pull/12 can be re forked to use the changes.

**Comment by Inrixia** [comment]
Updated to be upstream from https://github.com/alvgaona/cloudflare-discord-bot/pull/15

Please consider merging that first as after doing so code changes shown here will not also be showing formatting changes.

File: .prettierrc.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+	"useTabs": true,
```
Comment (alvgaona) [inline .prettierrc.json:2]: @Inrixia we should change it to `false`. Let's use spaces, not tabs. Shall we?

File: package.json (hunk 1 above)
Comment (alvgaona) [reply package.json:2]: Let's use spaces for this project. Can we make it 2 spaces?

**Review by alvgaona — CHANGES_REQUESTED** [review]

File: .prettierrc.json (hunk 5 above)
Comment (Inrixia) [reply .prettierrc.json:2]: Done

**Comment by Inrixia** [comment]
Please see https://github.com/alvgaona/cloudflare-discord-bot/pull/15 for the finalized prettier config with spaces etc. Merge that first then can do these

**Comment by Inrixia** [comment]
This is now upstream of the merged changes from https://github.com/alvgaona/cloudflare-discord-bot/pull/15 and ready to merge

**Review by alvgaona — APPROVED** [review]
Looking good now.

**Pull request merged**